### ISSUE TITLE
cv valid metric not identical to manual 

### ISSUE THREAD

**Issue by dtanalytic** [catboost/catboost#2338]
Problem:cv valid error not identical to manual 
catboost version:1.1.1
Operating System:Linux(google colab)
CPU:
GPU:
Please, help me to understand why cv and manual results give different auc metric. I intentionally dropped catgorical features from dataset (read that there tricks in working with them). My code example is below:

`import pandas as pd
import numpy as np
from sklearn.model_selection import KFold
from sklearn.metrics import roc_auc_score

from catboost import Pool, cv, CatBoostClassifier

SEED=0
np.random.seed(SEED)

df = pd.read_csv("https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv")
df = df.sample(frac=1).reset_index(drop=True)

id_cols = ['name', 'ticket']
target_col = 'survived'
num_cols = df.drop([target_col]+id_cols, axis=1).select_dtypes(include=np.number).columns.tolist()
df = df[[target_col]+id_cols+num_cols]

sp = KFold(n_splits=3, shuffle=False)

tr_pool = Pool(df.drop(columns=[target_col]+id_cols), df[target_col])

params = {'loss_function': 'Logloss', 'eval_metric':'AUC', 
          'verbose': False, 'metric_period':1, 'early_stopping_rounds':50}

res = cv(tr_pool, params = params, iterations=500, shuffle=False, folds=sp,stratified=False )
it_best = np.argmax(res['test-AUC-mean'])+1
metr_best = max(res['test-AUC-mean'])
print(f'it_best - {it_best}, metr_best - {metr_best}')

auc_l = []
clf = CatBoostClassifier(iterations=it_best, eval_metric='AUC', silent=True)
for idx_tr, idx_val in sp.split(df):
    
    X_tr, X_val = df.drop(columns=id_cols+[target_col]).iloc[idx_tr], \
                  df.drop(columns=id_cols+[target_col]).iloc[idx_val]
    y_tr, y_val = df[target_col].iloc[idx_tr], df[target_col].iloc[idx_val]
    
    
    clf.fit(X_tr, y_tr)
    
    auc_l.append(roc_auc_score(df[target_col].iloc[idx_val], clf.predict_proba(X_val)[:,1]))

print(np.mean(auc_l))`

output from cv - 0.7889263104273184, manual - 0.7826946255875921
